src/server/req_manager.c

```diff
@@ -1417,37 +1412,53 @@ void
 mgr_server_set(struct batch_request *preq, conn_t *conn)
 {
 	int	  bad_attr = 0;
+	svrattrl *tmp;
 	svrattrl *plist;
+	svrattrl *ulist;
+	pbs_list_head unsetlist;
 	int	  rc;
 
+	CLEAR_HEAD(unsetlist);
 
 	plist = (svrattrl *)GET_NEXT(preq->rq_ind.rq_manager.rq_attr);
 	/*Only root at server host can set server attribute "acl_roots".*/
 	while (plist) {
-		if (find_attr(svr_attr_def, plist->al_name, SRV_ATR_LAST) >= 0) {
-			if (strcasecmp(plist->al_name, ATTR_aclroot) == 0) {
-				if (!is_local_root(preq->rq_user, preq->rq_host)) {
-					reply_badattr(PBSE_ATTRRO, bad_attr, plist, preq);
-					return;
-				}
-				break;
+		if (strcasecmp(plist->al_name, ATTR_aclroot) == 0) {
+			if (!is_local_root(preq->rq_user, preq->rq_host)) {
+				reply_badattr(PBSE_ATTRRO, bad_attr, plist, preq);
+				return;
 			}
+		} else if (plist->al_atopl.value == NULL || plist->al_atopl.value[0] == '\0') {
```

Comment:
Just a question. Are we assuming that if a value for an attribute is not provided it is meant to be unset ?

Reply:
Yes, exactly. Earlier, we used to write the entire server structure directly, so any removed attributes from heap woudl also get "removed" from database. However, now we only update selected attributes. Thus, if we use this api to unset something (which is actually tested by some test cases), then we need to detect those and actually call unset_xxx routines to remove from database. I will add a comment to explain this.

Reply:
Thanks for the detailed explanation and adding the comment in the code.